xdsl/irdl.py

```diff
@@ -443,16 +443,25 @@ class RegionDef(Region):
     An IRDL region definition.
     """
 
+    def __init__(self):
```

Comment:
This became necessary for some reason...

Reply:
Ah, does it inherit the `init=False`? And the fields of Region now are not annotated with the stuff that makes dataclass derive the init, AFAIU.

Reply:
Maybe, although it still didn't work when I removed the init=False

Reply:
and the `field` annotations? I woudl expect them to be the issue. Can't we still have them, even if we override __init__?

Reply:
Good point, I'll give it a go